### PR TITLE
Add SqlPlugin sample

### PR DESCRIPTION
**Pull request by mslukebo**

### CODE REVIEW

**Review by trgibeau — CHANGES_REQUESTED** [review]

File: samples/SqlPlugin/SqlPlugin/SqlPlugin.csproj [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (mslukebo) [reply samples/SqlPlugin/SqlPlugin/SqlPlugin.csproj:5]: could be standard. Isn't netcore the cross-platform one though?

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlDataCooker.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        public override DataProcessingResult CookDataElement(SqlEvent sqlEvent,
+                                                             SqlSourceParser context,
+                                                             CancellationToken cancellationToken)
+        {
+            // Use context to get trace start time from the source parser's DataSourceInfo
+            var traceStartTime = context.DataSourceInfo.StartWallClockUtc;
+
+            var offset = sqlEvent.StartTime.Subtract(traceStartTime);
+            var relativeTimestamp = Timestamp.FromNanoseconds(offset.Ticks * 100);
```
[the remark this answers is not in the document]
Comment (mslukebo) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlDataCooker.cs:49]: Yes, the purpose of this cooker is to demonstrate mapping from a SqlEvent to a SqlEventWithRelativeTimestamp

File: samples/SqlPlugin/SqlPlugin/SqlPlugin.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (trgibeau) [reply samples/SqlPlugin/SqlPlugin/SqlPlugin.csproj:5]: both are

[the remark this answers is not in the document]
Comment (trgibeau) [reply samples/SqlPlugin/SqlPlugin/SqlPlugin.csproj:5]: .net standard opens more support

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlDataCooker.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (trgibeau) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlDataCooker.cs:49]: Sounds good!

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            // NOOP here; process everything :)
+        }
+
+        public void ProcessSource(ISourceDataProcessor<SqlEvent, SqlSourceParser, string> dataProcessor, ILogger logger, IProgress<int> progress, CancellationToken cancellationToken)
+        {
+            XNamespace ns = SqlPluginConstants.SqlXmlNamespace;
+
+            var xmlDocument = XDocument.Parse(File.ReadAllText(this.filePath));
+
+            var events = xmlDocument.Element(ns + "TraceData").Element(ns + "Events").Elements(ns + "Event");
```
[the remark this answers is not in the document]
Comment (mslukebo) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs]: I implemented the XMLReader... are you looking at an older commit?

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlTableFromDataCooker.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+using Microsoft.Performance.SDK.Extensibility;
+using Microsoft.Performance.SDK.Processing;
+using System;
+using System.Collections.Generic;
+
+namespace SqlPluginWithProcessingPipeline
+{
+    [Table]
+    public class SqlTableFromDataCooker
```
[the remark this answers is not in the document]
Comment (mslukebo) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlTableFromDataCooker.cs:12]: I noticed in our docs we were saying tables need to be static, but these tables are working without being declared static. Is this a hard requirement?

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (trgibeau) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs]: Weird, this is a old comment....

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlTableFromDataCooker.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (trgibeau) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlTableFromDataCooker.cs:12]: Its recommended b/c we don't use the table as a instance in the SDK

File: samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml.Linq;
+
+namespace SqlPluginWithProcessingPipeline
+{
+    public class SqlSourceParser
+            : ISourceParser<SqlEvent, SqlSourceParser, string>
```
[the remark this answers is not in the document]
Comment (mslukebo) [reply samples/SqlPlugin/SqlPluginWithProcessingPipeline/SqlSourceParser.cs:16]: I took a look at the base class, and it seems like the only thing I'd really simplify is letting me remove PrepareForProcessing and the `Type` properties. I'd rather show them in this sample.

**Review by trgibeau — APPROVED** [review]

**Pull request merged**